gulpfile.ts

```diff
@@ -1,46 +1,67 @@
-const execSync = require('child_process').execSync;
-
-const azureSDKForJSRepoRoot = __dirname;
-const azureRestAPISpecsRoot = args['azure-rest-api-specs-root'] || path.resolve(azureSDKForJSRepoRoot, '..', 'azure-rest-api-specs');
-const package = args['package'];
-const use = args['use'];
-const whatif = args['whatif'];
-
-function findReadmeTypeScriptMdFilePaths(azureRestAPISpecsRoot) {
-  const specificationFolderPath = path.resolve(azureRestAPISpecsRoot, 'specification');
-  return glob.sync('**/readme.typescript.md', { absolute: true, cwd: specificationFolderPath });
+import { execSync } from "child_process";
+import * as fs from "fs";
+import * as glob from "glob";
+import * as gulp from "gulp";
+import * as path from "path";
+import { argv } from "yargs";
+
+const azureSDKForJSRepoRoot: string = __dirname;
+const azureRestAPISpecsRoot: string = argv['azure-rest-api-specs-root'] || path.resolve(azureSDKForJSRepoRoot, '..', 'azure-rest-api-specs');
+const packageArg: string = argv['package'];
+const use: string = argv['use'];
+const whatif: boolean = argv['whatif'];
+
+function findReadmeTypeScriptMdFilePaths(azureRestAPISpecsRoot: string): string[] {
+  console.log(`Looking for "readme.typescript.md" files in "${azureRestAPISpecsRoot}"...`);
+  const specificationFolderPath: string = path.resolve(azureRestAPISpecsRoot, 'specification');
+  const readmeTypeScriptMdFilePaths: string[] = glob.sync('**/readme.typescript.md', { absolute: true, cwd: specificationFolderPath });
+  if (readmeTypeScriptMdFilePaths) {
+    for (let i = 0; i < readmeTypeScriptMdFilePaths.length; ++i) {
+      const readmeTypeScriptMdFilePath: string = readmeTypeScriptMdFilePaths[i];
+      console.log(`  Found "${readmeTypeScriptMdFilePath}".`);
+      if (readmeTypeScriptMdFilePath && !startsWith(readmeTypeScriptMdFilePath, specificationFolderPath)) {
+        const resolvedReadmeTypeScriptMdFilePath: string = path.resolve(specificationFolderPath, readmeTypeScriptMdFilePath);
+        console.log(`    Updating to "${resolvedReadmeTypeScriptMdFilePath}".`);
+        readmeTypeScriptMdFilePaths[i] = resolvedReadmeTypeScriptMdFilePath;
+      }
+    }
+  }
+  return readmeTypeScriptMdFilePaths;
 }
 
-function getPackageNameFromReadmeTypeScriptMdFileContents(readmeTypeScriptMdFileContents) {
+function getPackageNameFromReadmeTypeScriptMdFileContents(readmeTypeScriptMdFileContents: string): string {
   return readmeTypeScriptMdFileContents.match(/package-name: (\S*)/)[1];
 }
 
-function getOutputFolderFromReadmeTypeScriptMdFileContents(readmeTypeScriptMdFileContents) {
+function getOutputFolderFromReadmeTypeScriptMdFileContents(readmeTypeScriptMdFileContents: string): string {
   return readmeTypeScriptMdFileContents.match(/output-folder: (\S*)/)[1];
 }
 
-function npmInstall(packageFolderPath) {
-  execSync(`npm install`, { cwd: packageFolderPath, stdio: ['ignore', 'ignore', 'pipe'] });
+function npmInstall(packageFolderPath: string): any {
```

Comment:
is this return type used now?

Reply:
No. I added it just so that npmInstall() behaved a little more like execSync(), but since I wasn't using it I'll just make it void.